src/saltext/vmware/modules/dvswitch.py

```diff
@@ -281,3 +282,155 @@ def manage(
+
+        salt '*' vmware_dvswitch.remove_host switch_name=dvs1 hostname=host1
+
+    """
+    log.debug("Running vmware_dvswitch.remove_host")
+    ret = {}
+    if not service_instance:
+        service_instance = get_service_instance(opts=__opts__, pillar=__pillar__)
+    hosts = utils_esxi.get_hosts(
+        service_instance=service_instance,
+        host_names=[host_name] if host_name else None,
+        cluster_name=cluster_name,
+        datacenter_name=datacenter_name,
+        get_all_hosts=True if not host_name else False,
+    )
+    try:
+        for h in hosts:
+            ret[h.name] = False
+            if hasattr(h.configManager.networkSystem.networkInfo, "vswitch"):
+                h.configManager.networkSystem.RemoveVirtualSwitch(vswitchName=switch_name)
+                ret[h.name] = True
+        return ret
+    except (
+        vim.fault.InvalidState,
+        vim.fault.NotFound,
+        vim.fault.HostConfigFault,
+        vim.fault.ResourceInUse,
+        vmodl.fault.InvalidArgument,
+        salt.exceptions.VMwareApiError,
+    ) as exc:
+        raise salt.exceptions.SaltException(str(exc))
+
+
+def manage_host(
+    switch_name,
+    host_name=None,
+    datacenter_name=None,
+    cluster_name=None,
+    nics=None,
+    num_ports=None,
+    mtu=None,
+    service_instance=None,
+):
+    """
+    Add/update an ESXi host to a distributed vSwitch.
+
+    switch_name
+        Name of the distributed vSwitch.
+
+    datacenter_name
+        Filter by this datacenter name (required when cluster is specified)
+
+    cluster_name
+        Filter by this cluster name (optional)
+
+    host_name
+        Filter by this ESXi hostname (optional)
+
+    nics
+        List of vmnics to attach to vSwtich. (optional). Default "None".
+
+    num_ports
+        Number of ports to configure on the vSwitch. (optional). Default 128.
+
+    mtu
+        MTU to configure on the vSwitch. (optional). Default 1600.
+
+    service_instance
+        Use this vCenter service connection instance instead of creating a new one. (optional).
+
+    .. code-block:: bash
+
+        salt '*' vmware_esxi.manage_host switch_name=dvs1 host_name=host1 num_ports=256 mtu=1800
+
+    """
+    log.debug("Running vmware_dvswitch.manage_host")
+    ret = {}
+    if not service_instance:
+        service_instance = get_service_instance(opts=__opts__, pillar=__pillar__)
+    hosts = utils_esxi.get_hosts(
+        service_instance=service_instance,
+        host_names=[host_name] if host_name else None,
+        cluster_name=cluster_name,
+        datacenter_name=datacenter_name,
+        get_all_hosts=True if not host_name else False,
+    )
+    try:
+        for h in hosts:
+            ret[h.name] = False
+            network_manager = h.configManager.networkSystem
+            if not network_manager:
+                continue
+            if isinstance(nics, str):
+                nics = [nics]
+            vss_spec = vim.host.VirtualSwitch.Specification()
```

Comment:
VirtualSwitch is inside of host? This has to do something different from vim.dvs.VmwareDistributedVirtualSwitch... [I've found the docs for the latter](https://vdc-repo.vmware.com/vmwb-repository/dcr-public/9fd87c06-14a3-41e5-b28d-277864a80f29/d6112c2a-b124-4fa5-96d7-9fb4b6f1bb50/doc/vim.dvs.VmwareDistributedVirtualSwitch.html), but I'm having a little difficulty finding the docs for the former (vim.host.VirtualSwitch).

I'm really trying to understand what is going on here, because it seems so weird in the context of what I know about networking.

Reply:
good point, yes. it is a standard switch. wondering if I should move it to a vswitch module instead of this dvswitch.

Reply:
I don't have a sure way to do it. the switch_name here is a dvswitch and it lets the user configure params on the vswitch on the host, not the dvswitch param itself. So, I am going to leave it there in the dvswitch module.

Reply:
ok, that makes some things more clear

the dvswitch is a switch between regular virtual switches, this function creates a regular vswitch on the host, that is connected to the dvswitch

Reply:
right?